docs/guides/lint.md

```diff
@@ -3,75 +3,235 @@ redirectFrom:
+
+:::info
+Possible values specified above are related to built-in rules only. Custom plugins can contain additional configurations, so it can be as follows:
+
+```yaml
+lint:
+  extends:
+    - recommended
+    - my-custom-super-ruleset
+```
+
+Find more information in the [Configs in plugins](../resources/custom-rules/#configs-in-plugins) section.
+
+:::
+
+#### Examples
+
+```yaml single value
+lint:
   extends:
-    # This section is where you choose the base configurations.
-    # You may override specific settings in the subsequent sections.
-    - recommended # This is the default (and built in) configuration. If it is too strict, try `minimal`.
+    - minimal
+```
+
+```yaml multiple values
+lint:
+  extends:
+    - minimal
+    - recommended
+```
+
+### Resolve
+
+Use this section to specify external links in your definition that are not publicly accessible (except for Redocly API registry links).
+
+* **type**: `object`
+
+Redocly automatically resolves any API registry link or public URL. However, you may want to resolve links that are not API registry links or publicly accessible.
+
+Currently, OpenAPI CLI only supports `http` headers. Only one `http` header per URL is supported in the `resolve` section:
+
+```yaml
+lint:
+  resolve:
+    http:
+      headers:
+        - # header configuration
+```
+
+#### Header configuration
+
+| Property      | Description | Examples |
+| ------------- | ----------- | -------- |
+| `matches`     | Glob match against the URL. | `https://api.example.com/**` or `https://api.example.com/*.json` |
+| `name`        | HTTP header name.           | `Authorization` or `X-API-KEY` |
+| `value`       | The value of the header. Mutually exclusive with `envVariable`. | `123-abc` |
+| `envVariable` | The name of the environment variable that contains the value of the header. Mutually exclusive with `value`. | `SECRET_KEY` |
+
+:::success
+It is recommeded to use environment variables where possible.
+:::
+
+#### Example
+
+```yaml multiple values
+lint:
+  resolve:
+    http:
+      headers:
+        - matches: https://api.example.com/v2/**
+          name: X-API-KEY
+          envVariable: SECRET_KEY
+        - matches: https://example.com/*/test.yaml
+          name: Authorization
+          envVariable: SECRET_AUTH
+```
+
+The first match will win in the event when a URL matches multiple patterns. Therefore, only the header from the first match will be used in the request.
+
+### Preprocessors
+
+Use this section to change the [severity level](#severity-levels) of any rules in your extended configurations. Some rules may also receive [additional configurations](#additional-rule-options).
+
+* **type**: `array of objects`
+* **possible values**: `error`, `warn`, `off`
+
+Preprocessors run first during `lint` and `bundle`.
+
+:::info
+Linting happens only when the `--lint` flag is supplied
+:::
+
+:::info
+Preprocessors are rarely indicated - avoid if possible.
```

Comment:
question: what does "rarely indicated" mean in this case? I can't figure out if we're telling the users to avoid specifying any preprocessors, or to avoid using them altogether...?